src/converse-rosterview.js

```diff
@@ -354,10 +354,20 @@ converse.plugins.add('converse-rosterview', {
                     u.hideElement(this.el);
                     return this;
                 }
-                const ask = this.model.get('ask'),
-                    show = this.model.presence.get('show'),
-                    requesting  = this.model.get('requesting'),
-                    subscription = this.model.get('subscription');
+
+                const contact_jid = this.model.get('jid'),
+                    requesting = this.model.get('requesting'),
+                    subscription = u.isSameBareJID(contact_jid, _converse.connection.jid) ? 'both' : this.model.get('subscription'),
```

Comment:
You shouldn't do this.

Instead, when the contact gets created, check whether it's yourself and then if so, set `subscription` as `both`, so that the value is persisted.

However, I'm not sure this is right in general, the right value must be stored on the XMPP server, then we don't need any checks like this in Converse and it'll also work properly on other clients.

So I think it's still better to let yourself show up as a requesting contact and then to accept your own contact request, then doing it like this here.